portfolio/src/main/webapp/script.js

```diff
@@ -98,10 +98,173 @@ function setInfoEvents() {
+ */
+function handleGivenComments(comments) {
+  for (var i = 0; i < comments.length; i++) {
+    addCommentToDom(comments[i]);
+  }
+}
+
+const COMMENTS_DISPLAY = "commentsDisplay";
+
+/**
+ * Adds the comment to the page
+ */
+function addCommentToDom(comment) {
+  const commentDisplay = document.getElementById(COMMENTS_DISPLAY);
+  const obj = makeCommentElement
+    (comment, comment.commentText, comment.commentAuthor);
+
+  // add the new objects to the comment Display
+  commentDisplay.appendChild(obj);
+}
+
+/**
+ * @return HTML div object containing the comment with relevant information and styling
+ */
+function makeCommentElement(comment, text, author) {
+  const commentObj = document.createElement("div");
+
+  /**Class name to style all comment blocks */
+  const COMMENT_CLASS = "comment";
+
+  commentObj.classList.add(COMMENT_CLASS);
+  
+  commentObj.appendChild(makeDeleteButton(comment, commentObj));
+  commentObj.appendChild(makeCommentAuthorElement(author));
+  commentObj.appendChild(makeCommentTextElement(text));
+
+  return commentObj;
+}
+
+function makeDeleteButton(comment, commentElement) {
+  const button = document.createElement('div');
+  button.innerHTML = '<i class="far fa-trash-alt"></i>';
+  button.classList.add("deleteButton");
+  button.addEventListener('click', () => {
+    deleteComment(comment);
+    commentElement.remove();
+  });
+
+  return button;
+}
+
+/**
+ * @return HTML paragraph containing the comment text
+ */
+function makeCommentTextElement(text) {
+  const commentText = document.createElement("p");
+  commentText.innerText = `"${text}"`;
+
+  /**Class name to style the comment text blocks */
+  const COMMENT_TEXT = "commentText";
+  commentText.classList.add(COMMENT_TEXT);
+
+  return commentText;
+}
+
+/**
+ * @return HTML paragraph containing the author's name
+ */
+function makeCommentAuthorElement(author) {
+  const commentAuthor = document.createElement("p");
+  commentAuthor.innerText = `@${author}`;
+
+  /**Class name to style the comment author name */
+  const COMMENT_AUTHOR = "commentAuthor";
+  commentAuthor.classList.add(COMMENT_AUTHOR);
+
+  return commentAuthor;
+}
+
+/**
+ * Inserts new comment into the comment Display
+ */
+function updateComments() {
+  var commentText = getFormValue("inputComment");
+  var commentAuthor = getFormValue("inputName");
+  if (validComment(commentText)) {
+    const params = new URLSearchParams();
+    params.append('comment-text', commentText);
+    params.append('comment-author', commentAuthor);
+    fetch('new-comment', {method: 'POST', body: params}).then(refreshComments);
+  }
+}
+
+function refreshComments() {
+    clearComments();
+    getComments();
+}
+
+/**
+ * @return If the comment is not blank
+ */
+function validComment(comment) {
+  return !(comment.length === 0);
+}
+
+/**
+ * Gets and clears the form input element
+ */
+function getFormValue(id) {
```

Comment:
nit: It can be confusing for a getter to have a side-effect, like clearing the value. I'd recommend renaming this to something like `clearFormValue()` and make it return the previous value as you did.

Reply:
That makes sense, I guess it's similar to how most built-in remove or delete functions return the object being removed.